playbooks/ci-full-centos/tasks/pre-deploy.yml

```diff
@@ -0,0 +1,98 @@
+---
+- name: controller 0 hosts file
+  hosts: controller[0]
+  vars_files:
+  - ../vars/main.yml
+  tasks:
+    - name: add controller node to /etc/hosts
+      lineinfile: dest=/etc/hosts regexp={{ testenv_instance_names[0] }}
+                  insertafter=EOF
+                  line="127.0.1.1 {{ testenv_instance_names[0] }}"
+
+- name: controller 1 hosts file
+  hosts: controller[1]
+  vars_files:
+  - ../vars/main.yml
+  tasks:
+    - name: add controller node to /etc/hosts
+      lineinfile: dest=/etc/hosts regexp={{ testenv_instance_names[1] }}
+                  insertafter=EOF
+                  line="127.0.1.1 {{ testenv_instance_names[1] }}"
+
+- name: compute 0 hosts file
+  hosts: compute[0]
+  vars_files:
+  - ../vars/main.yml
+  tasks:
+    - name: add compute node to /etc/hosts
+      lineinfile: dest=/etc/hosts regexp={{ testenv_instance_names[2] }}
+                  insertafter=EOF
+                  line="127.0.1.1 {{ testenv_instance_names[2] }}"
+
+- name: tasks for all
+  hosts: all
+  serial: 10
+  tasks:
+  # Set Locale
+  - name: Generate en_US locale
+    command: localedef -c -f UTF-8 -i en_US en_US.UTF-8
+
+  - name: add pip
+    yum: name={{item}} state=installed
+    with_items:
+      - epel-release
+    register: result
+    until: result|succeeded
+    retries: 5
+
+  - name: add pip
+    yum: name={{item}} state=installed
+    with_items:
+      - python-pip
+    register: result
+    until: result|succeeded
+    retries: 5
+
+  - name: get backports.ssl_match_hostname for apt repo installs
+    pip:
+      name: backports.ssl_match_hostname
+    register: result
+    until: result|succeeded
+    retries: 5
+
+  #- name: Add BlueBox OpenStack PPA
+  #  apt_repository: repo='ppa:blueboxgroup/openstack'
+  #  register: result
+  #  until: result|succeeded
+  #  retries: 5
+
+  # don't run this on existing clusters
+  - name: Update all packages
+    yum: name=* state=latest
+    register: result
+    until: result|succeeded
+    retries: 5
+
+  #- name: Ubuntu 14.04 style /etc/network/interfaces.d/ directory
+  #  file: dest=/etc/network/interfaces.d state=directory owner=root
+  #        group=root mode=0755
+
+  #- name: /etc/network/interfaces to support interfaces.d/
+  #  template: src=../templates/etc/network/interfaces
+  #            dest=/etc/network/interfaces
+
+  #- name: default interfaces.d/eth0.cfg
+  #  template: src=../templates/etc/network/interfaces.d/eth0.cfg
+  #            dest=/etc/network/interfaces.d/eth0.cfg
+
+#- name: network tasks
+#  hosts: network
+#  tasks:
+#  - name: interfaces.d/eth0.cfg with NAT for floating IP pool
+#    template: src=../templates/etc/network/interfaces.d/eth0-controllers.cfg
+#              dest=/etc/network/interfaces.d/eth0.cfg
+#    notify: bounce eth0
+#
+#  handlers:
+#  - name: bounce eth0
+#    shell: ifdown eth0; ifup eth0
```

Comment:
I'm assuming this needs to be enabled in a RedHat-y way, but I did not have time to figure it out.